The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6676385622/job/18145374013) [(plain)](https://github.com/rust-lang/rust/commit/dbdece8854c3ebd84cdf5d354d289c295b93f63b/checks/18145374013/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_b12406f3-b41f-43eb-adfe-2dd4f01340e7
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=cg_gcc-tests
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_b12406f3-b41f-43eb-adfe-2dd4f01340e7
GITHUB_REF=refs/pull/117313/merge
GITHUB_REF_NAME=117313/merge
GITHUB_REF_PROTECTED=false
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-15', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-missing-tools']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-15/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: rust.codegen-units-std := 1
---
   Compiling tempfile v3.7.1
   Compiling rustc_codegen_gcc v0.1.0 (/checkout/compiler/rustc_codegen_gcc)
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" VSLANG="1033" "cc" "-Wl,--version-script=/tmp/rustcHlXT6o/list" "-Wl,--no-undefined-version" "-m64" "/tmp/rustcHlXT6o/symbols.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.00.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.01.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.02.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.03.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.04.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.05.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.06.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.07.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.08.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.09.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.10.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.11.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.12.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.13.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.14.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.rustc_codegen_gcc.359752fbeaa460ef-cgu.15.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/rustc_codegen_gcc-56a1fc03646ac2c3.3i03xy6tyakqaf4a.rcgu.rmeta" "-Wl,--as-needed" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/release/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libsmallvec-715d6280ffa5c766.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libtempfile-27b9fed622356b87.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libfastrand-d77c7684ca21d5fc.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/librustix-2141b0131335e511.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libbitflags-759cbb9c48597520.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/liblinux_raw_sys-cbe921d69978c5fe.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libcfg_if-c76496b37c5c4175.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libobject-ec880f1cec7e49b7.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libmemchr-31aa04cc2908fb80.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libgccjit-3c2f17fc7f1c348e.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/libgccjit_sys-2378bde9594993f7.rlib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/liblibc-07d1250a9f0eb185.rlib" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bdynamic" "-lrustc_driver-8722abd79f66efd0" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-lstd-eaaccbfd3dfc37e3" "-Wl,-Bstatic" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-a943d66d0644a9a0.rlib" "-Wl,-Bdynamic" "-lgccjit" "-lLLVM-15" "-ldl" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-sysroot/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-codegen/x86_64-unknown-linux-gnu/release/deps/librustc_codegen_gcc-56a1fc03646ac2c3.so" "-shared" "-Wl,-z,relro,-z,now" "-Wl,-O1" "-nodefaultlibs" "-Wl,-z,origin" "-Wl,-rpath,$ORIGIN/../lib"
  = note: /usr/bin/ld: cannot find -lgccjit: No such file or directory
          collect2: error: ld returned 1 exit status

error: could not compile `rustc_codegen_gcc` (lib) due to previous error
Build completed unsuccessfully in 0:03:21
  local time: Sat Oct 28 10:38:59 UTC 2023
```

</details>

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -57,8 +57,9 @@ jobs:
             os: ubuntu-20.04-4core-16gb
             env: {}
           - name: x86_64-gnu-llvm-15
+            env:
+              SKIP_CODEGEN_TESTS: "1"
             os: ubuntu-20.04-16core-64gb
-            env: {}
           - name: x86_64-gnu-tools
             os: ubuntu-20.04-16core-64gb
             env: {}

--- a/src/ci/docker/host-x86_64/x86_64-gnu-llvm-15/Dockerfile
+++ b/src/ci/docker/host-x86_64/x86_64-gnu-llvm-15/Dockerfile
@@ -37,6 +37,8 @@ RUN sh /scripts/sccache.sh
 
 # Make `libgccjit.so` accessible.
 RUN ln -s /usr/lib/gcc/x86_64-linux-gnu/12/libgccjit.so /usr/lib/x86_64-linux-gnu/libgccjit.so
+# Fix rustc_codegen_gcc lto issues.
+ENV GCC_EXEC_PREFIX="/usr/lib/gcc/"
 
 # We are disabling CI LLVM since this builder is intentionally using a host
 # LLVM, rather than the typical src/llvm-project LLVM.

--- a/src/ci/docker/host-x86_64/x86_64-gnu-llvm-15/script.sh
+++ b/src/ci/docker/host-x86_64/x86_64-gnu-llvm-15/script.sh
@@ -4,7 +4,11 @@ set -ex
 
 # Only run the stage 1 tests on merges, not on PR CI jobs.
 if [[ -z "${PR_CI_JOB}" ]]; then
-    ../x.py --stage 1 test --skip src/tools/tidy --skip tests/codegen
+    if [[ "${SKIP_CODEGEN_TESTS}" == "1" ]]; then
+        ../x.py --stage 1 test --skip src/tools/tidy --skip tests/codegen
+    else
+        ../x.py --stage 1 test --skip src/tools/tidy
+    fi
 
     # Run the `mir-opt` tests again but this time for a 32-bit target.
     # This enforces that tests using `// EMIT_MIR_FOR_EACH_BIT_WIDTH` have
@@ -20,7 +24,11 @@ if [[ -z "${PR_CI_JOB}" ]]; then
 fi
 
 # NOTE: intentionally uses all of `x.py`, `x`, and `x.ps1` to make sure they all work on Linux.
-../x.py --stage 2 test --skip src/tools/tidy --skip tests/codegen
+if [[ "${SKIP_CODEGEN_TESTS}" == "1" ]]; then
+    ../x.py --stage 2 test --skip src/tools/tidy --skip tests/codegen
+else
+    ../x.py --stage 2 test --skip src/tools/tidy
+fi
 
 # Run the `mir-opt` tests again but this time for a 32-bit target.
 # This enforces that tests using `// EMIT_MIR_FOR_EACH_BIT_WIDTH` have

--- a/src/ci/docker/host-x86_64/x86_64-gnu-tools/Dockerfile
+++ b/src/ci/docker/host-x86_64/x86_64-gnu-tools/Dockerfile
@@ -64,6 +64,8 @@ RUN sh /scripts/sccache.sh
 
 # Make `libgccjit.so` accessible.
 RUN ln -s /usr/lib/gcc/x86_64-linux-gnu/12/libgccjit.so /usr/lib/x86_64-linux-gnu/libgccjit.so
+# Fix rustc_codegen_gcc lto issues.
+ENV GCC_EXEC_PREFIX="/usr/lib/gcc/"
 
 COPY host-x86_64/x86_64-gnu-tools/checktools.sh /tmp/
 

--- a/src/ci/docker/run.sh
+++ b/src/ci/docker/run.sh
@@ -271,6 +271,7 @@ docker \
   run \
   --workdir /checkout/obj \
   --env SRC=/checkout \
+  --env "SKIP_CODEGEN_TESTS=$SKIP_CODEGEN_TESTS" \
   $args \
   --env CARGO_HOME=/cargo \
   --env DEPLOY \

--- a/src/ci/github-actions/ci.yml
+++ b/src/ci/github-actions/ci.yml
@@ -332,6 +332,8 @@ jobs:
 
           - name: x86_64-gnu-llvm-15
             <<: *job-linux-16c
+            env:
+              SKIP_CODEGEN_TESTS: "1"
 
           - name: x86_64-gnu-tools
             <<: *job-linux-16c

--- a/src/ci/run.sh
+++ b/src/ci/run.sh
@@ -126,8 +126,13 @@ else
 
   RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set rust.verify-llvm-ir"
 
-  # Test the Cranelift backend in CI. Bootstrap knows which targets to run tests on.
-  RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set rust.codegen-backends=llvm,cranelift,gcc"
+  if [[ "${SKIP_CODEGEN_TESTS}" == "1" ]]; then
+    # Test the Cranelift backend in CI. Bootstrap knows which targets to run tests on.
+    RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set rust.codegen-backends=llvm,cranelift"
+  else
+    # Test the Cranelift and GCC backends in CI. Bootstrap knows which targets to run tests on.
+    RUST_CONFIGURE_ARGS="$RUST_CONFIGURE_ARGS --set rust.codegen-backends=llvm,cranelift,gcc"
+  fi
 
   # We enable this for non-dist builders, since those aren't trying to produce
   # fresh binaries. We currently don't entirely support distributing a fresh

